lib/internal/errors.js

```diff
@@ -286,6 +286,46 @@ function uvException(ctx) {
   return err;
 }
 
+/**
+ * This creates an error compatible with errors produced in the C++
+ * This function should replace the deprecated
+ * `exceptionWithHostPort(err, syscall, address, port, additional)` function.
+ *
+ * @param {number} err - A libuv error number
+ * @param {string} syscall
+ * @param {string} address
+ * @param {number} [port]
+ * @param {string} [additional]
+ * @returns {Error}
+ */
+function uvExceptionWithHostPort(err, syscall, address, port, additional) {
+  const [ code, uvmsg ] = errmap.get(err);
+  const message = `${syscall} ${code}: ${uvmsg}`;
+  let details = '';
+
+  if (port && port > 0) {
+    details = ` ${address}:${port}`;
+  } else if (address) {
+    details = ` ${address}`;
+  }
+  if (additional) {
+    details += ` - Local (${additional})`;
```

Comment:
What is this bit for? It seems to be unused, and the purpose is not 100 % obvious to me…

Reply:
This is a backwards compatibility from the deprecated `exceptionWithHostPort` function.
I took this logic directly from this function in order to keep the same string manipulation.